Src/POI.DiscordDotNet/Services/Implementations/DiscordStarboardService.cs

```diff
@@ -0,0 +1,128 @@
+﻿using DSharpPlus;
+using DSharpPlus.Entities;
+using DSharpPlus.EventArgs;
+using Microsoft.Extensions.Logging;
+using POI.Persistence.Domain;
+using POI.Persistence.Repositories;
+
+namespace POI.DiscordDotNet.Services.Implementations;
+
+public class DiscordStarboardService : IAddDiscordClientFunctionality
+{
+	private readonly ILogger<DiscordChatCommandsService> _logger;
+	private readonly IStarboardMessagesRepository _starboardMessagesRepository;
+	private readonly IServerSettingsRepository _serverSettingsRepository;
+
+	public DiscordStarboardService(
+		ILogger<DiscordChatCommandsService> logger,
+		IStarboardMessagesRepository starboardMessagesRepository, IServerSettingsRepository serverSettingsRepository)
+	{
+		_logger = logger;
+		_starboardMessagesRepository = starboardMessagesRepository;
+		_serverSettingsRepository = serverSettingsRepository;
+	}
+
+	public void Setup(IDiscordClientProvider discordClientProvider)
+	{
+		_logger.LogDebug("Setting up DiscordStarboardService");
+
+		var client = discordClientProvider.Client!;
+		client.MessageReactionAdded -= OnReactionAdded;
+		client.MessageReactionAdded += OnReactionAdded;
+	}
+
+	public void Cleanup(IDiscordClientProvider discordClientProvider)
+	{
+		_logger.LogDebug("Cleaning up DiscordStarboardService");
+		var client = discordClientProvider.Client!;
+		client.MessageReactionAdded -= OnReactionAdded;
+	}
+
+	private async Task OnReactionAdded(DiscordClient sender, MessageReactionAddEventArgs args)
+	{
+		//TODO: Check if message was not cached
+		//TODO: Check on emoji ID, not name
+		//TODO: Figure out a way to get the display name of the user outside the guild...
+
+		var serverSettings = await _serverSettingsRepository.FindOneById(args.Channel.Guild.Id);
+		if (serverSettings?.StarboardChannelId == null)
+		{
+			_logger.LogError("Server settings or starboard channel id not found!");
+			return;
+		}
+
+		// Skip event if the message is in the starboard channel (To prevent people staring the bot messages)
+		if(args.Channel.Id == serverSettings.StarboardChannelId)
+		{
+			return;
+		}
+
+		// Check if the message reactions contains the star emote
+		if (args.Message.Reactions.All(x => x.Emoji.Name != "⭐"))
```

Comment:
Preferably introduce a `var message = args.Message` to reduce the repeated property calls.